pkg/ast/spl/spl.peg

```diff
@@ -1386,6 +1402,10 @@ Aggregator <- aggFunc:AggFunction asField:AsField? {
     return agg, nil
 }
 
+// For evaluation statements inside aggregate functions:
+// 1. min, max, sum, avg, and range can only evaluate boolComparisonExpr, as they need to compute the field value inside the expression. There should be only one field.
+// 2. count can evaluate boolExpr because it does not care about the field value inside the expression. It only cares about whether the current row should be counted or not.
+// 3. distinctCount and values can evaluate valueExpr because they just need to find out the distinct values of the result of the expression. As for the type of the expression result, it is not important.
```

Comment:
I google it but I can only find examples in here [Use stats with eval expressions and functions](https://docs.splunk.com/Documentation/SplunkCloud/9.1.2308/Search/Usestatswithevalexpressionsandfunctions)

In the first example, 
`status=* | stats dc(eval(if(status=404, clientip, NULL()))) AS dc_ip_errors`

In here, you can see the eval statements will return a string or number value. Which might has some difference with min, max, etc. Since they just support `boolComparisonExpr`